File: problems/problems_LCR_002/solution.ts
function addBinary(a: string, b: string): string {
	if (a.length < b.length) {
		return addBinary(b, a);
	}
	const result: string[] = [];
	let carry: number = 0;
	const d: number = a.length - b.length;
	for (let i: number = a.length - 1; i >= 0; i--) {
		const sum: number = parseInt(a[i]) + (i - d >= 0 ? parseInt(b[i - d]) : 0) + carry;
		result.push((sum % 2).toString());
		carry = Math.floor(sum / 2);
	}
	if (carry) {
		result.push("1");
	}
	return result.reverse().join("");
};

export function Solve(inputJsonElement: string): any {
	const inputValues: string[] = inputJsonElement.split("\n");
	const a: string = JSON.parse(inputValues[0]);
	const b: string = JSON.parse(inputValues[1]);
	return addBinary(a, b);
}

File: problems/problems_LCR_002/solution.rs
#![allow(non_snake_case)]
use serde_json::{json, Value};

pub struct Solution;

impl Solution {
    pub fn add_binary(a: String, b: String) -> String {
		if a.len() < b.len() {
			return Solution::add_binary(b, a);
		}
		let mut res: Vec<char> = Vec::new();
		let mut carry = 0;
		let n = a.len();
		let d = n - b.len();
		let chars_a: Vec<char> = a.chars().collect();
		let chars_b: Vec<char> = b.chars().collect();
		for i in (0..n).rev() {
			let mut sum = carry;
			if i >= d {
				if chars_b[i - d] == '1' {
					sum += 1;
				}
			}
			if chars_a[i] == '1' {
				sum += 1;
			}
			carry = sum / 2;
			res.push(std::char::from_digit(sum % 2, 10).unwrap());
		}
		if carry > 0 {
			res.push('1');
		}
		res.iter().rev().collect()
    }
}

#[cfg(feature = "solution_LCR_002")]
pub fn solve(input_string: String) -> Value {
	let input_values: Vec<String> = input_string.split('\n').map(|x| x.to_string()).collect();
	let a: String = serde_json::from_str(&input_values[0]).expect("Failed to parse input");
	let b: String = serde_json::from_str(&input_values[1]).expect("Failed to parse input");
	json!(Solution::add_binary(a, b))
}
